src/fetchtastic/download/firmware.py

```
@@ -2062,6 +2142,90 @@ def sort_key(directory: str) -> Tuple:
+        candidate_dirs: list[str],
+        history_entries: list[dict[str, Any]],
+    ) -> Optional[str]:
+        """Select the latest prerelease directory using commit-history chronology.
+
+        Firmware repo prereleases are distinguished by a commit-hash suffix, not
+        by semver prerelease components.  Therefore the "latest" prerelease must
+        be chosen by repository commit chronology (added_at), not by hash-string
+        ordering or VersionManager.get_release_tuple.
+
+        Ranking (higher wins):
+          1. has_history – entry present in history_entries as active
```

Ruff (0.15.12)

[warning] Docstring contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF002)